src/js/other-websites/inpage_menu.js

```diff
@@ -51,28 +51,386 @@ async function isUserSignedIn() {
+      });
+
+      if (maxNumAliasesReached) {
+        generateAliasBtn.remove();
+        sendInPageEvent("viewed-menu", "input-menu-max-aliases-message");
+        remainingAliasesSpan.textContent = browser.i18n.getMessage(
+          "pageNoMasksRemaining"
+        );
+
+        getUnlimitedAliasesBtn.classList.remove("t-secondary");
+        getUnlimitedAliasesBtn.classList.add("t-primary");
+        // Focus on "Get unlimited alias" button
+        getUnlimitedAliasesBtn.focus();
+
+        document.querySelector(".fx-relay-menu-masks-lists").style.order = "2";
+      } else {
+        // Focus on "Generate New Alias" button
+        generateAliasBtn.focus();
+      }
+
+      //Check if premium features are available
+      // const premiumCountryAvailability = (
+      //   await browser.storage.local.get("premiumCountries")
+      // )?.premiumCountries;
+
+      // if (
+      //   premiumCountryAvailability?.premium_available_in_country !== true ||
+      //   !maxNumAliasesReached
+      // ) {
+      //   getUnlimitedAliasesBtn.remove();
+      // }
+
+      fxRelayMenuBody.classList.remove("is-loading");
+      // User is signed in/free: Remove the premium section from DOM so there are no hidden/screen readable-elements available
+      fxRelayMenuBody.classList.remove("is-premium");
+      signedInContentFree?.classList.remove("is-hidden");
+      signedInContentPremium?.remove();
+
+      await browser.runtime.sendMessage({
+        method: "updateIframeHeight",
+        height: fxRelayMenuBody.scrollHeight,
+      });
+    },
+    premium: async () => {
+      console.log("loggedIn/premium");
+
+      const fxRelayMenuBody = document.getElementById("fxRelayMenuBody");
+
+      const signedInContentFree = document.querySelector(
+        ".fx-content-signed-in-free"
+      );
+
+      const signedInContentPremium = document.querySelector(
+        ".fx-content-signed-in-premium"
+      );
+
+      // User is signed in/premium: Remove the free section from DOM so there are no hidden/screen readable-elements available
+      signedInContentPremium?.classList.remove("is-hidden");
+      signedInContentFree?.remove();
+
+      const searchInput = document.querySelector(
+        ".fx-relay-menu-masks-search-input"
+      );
+
+      searchInput.placeholder = browser.i18n.getMessage("labelSearch");
+
+      // Resize iframe
+      await browser.runtime.sendMessage({
+        method: "updateIframeHeight",
+        height: fxRelayMenuBody.scrollHeight,
+      });
+
+      const searchResults = document.querySelector(
+        ".fx-relay-menu-masks-search-results"
+      );
+      const searchResultsList = searchResults.querySelector("ul");
+
+      // Check if user may have custom domain masks
+      const premiumSubdomainSet = await browser.storage.local.get(
+        "premiumSubdomainSet"
+      );
+
+      const masks = await getMasks({
+        subdomainSet: premiumSubdomainSet.hasOwnProperty("premiumSubdomainSet"),
```

Comment:
Would this not be `true` if `premiumSubdomainSet` is `false`? (Also, small suggestion to use `is*` for boolean variable names - I first thought that this would be a set of subdomains...)

Reply:
The response from this request is a string set to what the subdomain is. I used the same strategy we're using here to validate if a user has a subdomain registered: https://github.com/mozilla/fx-private-relay-add-on/blob/mpp-830-sign-back-in-page/src/js/popup/popup.js#L198-L200

…but now that I just referenced that. I see now how we're determining "false". I'll work on this.

Reply:
Revised in 976f8762ea5f36197be222b7e9101921e0c5b172